### PR TITLE
Pulled in Ellas changes from CR-221

### PR DESCRIPTION
**Pull request by andrewjohnysons**
Refactored
       Added address tests
       Added Spring boot for RestTemplate.
       Changed debug levels and added logback.xml
       Beefed up the tests and fixed a missing message
       Added tests for invalid postcodes
       removed old tests and cleaned up
       readme file
       Logging
       Added URL configuration CC_BASE_URL

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java [hunk 1]
```diff
@@ -13,10 +13,14 @@
 import static org.junit.Assert.*;
 
 public class TestAddressEndpoints extends SpringIntegrationTest {
+    protected String ccBaseUrl  = System.getenv("CC_BASE_URL") == null ?
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java]: ? not using contact-centre-settings.host and port

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -106,3 +105,9 @@ queueconfig:
   
 report-settings:
   cron-expression: "0 * * * * *"
+
+contact-centre-settings:
+    username: serco_cks
+    password: temporary
+    host: ${CC_BASE_URL:localhost}
```
Comment (philwhiles) [inline src/main/resources/application.yml]: Used? This and TestAddressEndpoints code don't seem to marry

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java [hunk 3]
```diff
@@ -13,10 +13,14 @@
...
+            "http://localhost:8171" : System.getenv("CC_BASE_URL");
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java]: application properties implies host and port config'd separately, but used where? and this code implies BASE_URL is combo protocol://host:port
?
?

**Comment by andrewjohnysons** [comment]
I've updated this to use the application-test.yml file properly now and the application takes the @Value annotation to configure url, port, username, password

You will need environment variables:
CONTACT_CENTRE_SERVICE_URL
CONTACT_CENTRE_SERVICE_PORT

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java [hunk 4]
```diff
@@ -49,6 +55,7 @@ public void i_have_an_invalid_Postcode(String postcode) {
 
     @When("I Search Addresses By Invalid Postcode")
     public void i_Search_Addresses_By_Invalid_Postcode() {
+        final String postcodeUrl = ccBaseUrl + ":" + ccBasePort + "/addresses/postcode";
         RestTemplate restTemplate = new RestTemplateBuilder().basicAuthentication("serco_cks", "temporary").build();
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/TestAddressEndpoints.java]: hardwired creds should be replaced by app yaml @values

**Pull request merged**